Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,426 @@
+            }
+            //less than 5 seconds to delam, count every second.
+            else if (RoundSeconds <  5 && SMcomponent.SpeakAccumulator >= 1)
+            {
+                SMcomponent.SpeakAccumulator -= 1;
+                _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-seconds-before-delam", ("Seconds", RoundSeconds)), "Supermatter", false);
+            }
+
+            //play an alarm as long as you're delaming
+            if (SMcomponent.FinalCountdown)
+            {
+                SoundSystem.Play(Filter.Pvs(Uid, 5), SMcomponent.DelamAlarm.GetSound(), Uid, audioParams);
+            }
+
+            //TODO: make tesla(?) spawn at SupermatterComponent.PowerPenaltyThreshold and think up other delam types
+            //times up, explode or make a singulo
+            if (SMcomponent.DelamTimerAccumulator >= SupermatterComponent.DelamTimerTimer)
+            {
+                if (_delamType.Equals(SupermatterComponent.DelamType.Singulo))
+                {
+                    //spawn a singulo :)
+                    EntityManager.SpawnEntity("Singularity", Xform.Coordinates);
+                }
+                else if (_delamType.Equals(SupermatterComponent.DelamType.Explosion))
+                {
+                    //esplosion!!!!!
+                    _explosion.TriggerExplosive(Uid, explosive: Xplode, totalIntensity: 500000, radius: 500, user: Uid);
+                }
+
+                SMcomponent.FinalCountdown = false;
+                return;
+            }
+        }
+
+        /// </summary>
+        /// Determines if an entity can be dusted
+        /// </summary>
+        public bool CannotDestroy(EntityUid Uid)
+        {
+            bool Tag = false;
+            bool Static = false;
+
+            if (EntityManager.HasComponent<TagComponent>(Uid))
+            {
+                Tag = _tag.HasTag(Uid, "SMimmune");
+            }
+
+            if (EntityManager.HasComponent<PhysicsComponent>(Uid))
+            {
+                Static = EntityManager.GetComponent<PhysicsComponent>(Uid).BodyType.ToString() == "Static";
+            }
+
+            return Tag || Static;
+        }
+
+        private void OnCollideEvent(EntityUid uid, SupermatterComponent supermatter, StartCollideEvent args)
+        {
+            EntityUid Target = args.OtherFixture.Body.Owner;
+            if (!supermatter.Whitelist.IsValid(Target) || CannotDestroy(Target) || _container.IsEntityInContainer(uid)) return;
+
+            if (EntityManager.TryGetComponent<SupermatterFoodComponent>(Target, out var SupermatterFood))
+            {
+                supermatter.Power += SupermatterFood.Energy;
+            }
+            else if (EntityManager.TryGetComponent<ProjectileComponent>(Target, out var Projectile))
+            {
+                supermatter.Power += (float) Projectile.Damage.Total;
```

Comment:
This should probably scale with some value. (It does on the VG version and I agree).